base_layer/core/src/transactions/transaction.rs

```diff
@@ -281,6 +282,12 @@ impl UnblindedOutput {
     }
 
     pub fn as_transaction_output(&self, factories: &CryptoFactories) -> Result<TransactionOutput, TransactionError> {
+        if factories.range_proof.range() < 64 && self.value > MicroTari::from(1u64.shl(&factories.range_proof.range()))
```

Comment:
I think this is correct as opposed to `shl(....range() - 1)`

Reply:
`factories.range_proof.range() < 64` will never be true in our case since we set the range to 64. 
`1u64 << 64` [will overflow](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a44daa637b2c86d392624a76d9653d98) so should be `range() - 1`.

Reply:
it's actually true for 90% of the tests, that's why it's there. My reasoning is that if it is 64, then a u64 automatically fits in. 

The reason that this is here is because there are some tests that use a smaller range of 32. I'll have to check and see if `range() - 1` is correct in this case

Reply:
so just trying `1u64.shl(64)` panics with `thread 'main' panicked at 'attempt to shift left with overflow'`.

1u64(63) will give you half the 64-bit range. 

If you are looking for the max value of an arbitrary bit depth you will need to shift up that many ones.

Reply:
This is correct now that I've added the `=` to the `>=`. The case where the range is 64 is covered simply by the fact that you cannot include a higher value inside a u64.

Reply:
I guess the short circuit protects us from the overflow